lib/hammer_cli_foreman_virt_who_configure/config.rb

```diff
@@ -0,0 +1,222 @@
+require 'hammer_cli_foreman_virt_who_configure/system_caller'
+
+module HammerCLIForemanVirtWhoConfigure
+  MODE_UNLIMITED = 0
+  MODE_WHITELIST = 1
+  MODE_BLACKLIST = 2
+
+  class VirtWhoConfig < HammerCLIForeman::Command
+    resource :configs
+
+    def self.format_interval(interval)
+      hr_interval = (interval / 60)
+      if hr_interval <= 1
+        _('every hour')
+      else
+        _('every %s hours') % (interval / 60)
+      end
+    end
+
+    def self.format_status(status)
+      case status
+        when 'unknown'
+          _('No Report Yet')
+        when 'ok', 'out_of_date'
+          _('OK')
+        when 'error'
+          _('Error')
+        else
+          _('Unknown configuration status')
+      end
+    end
+
+    def self.format_filtering_mode(mode)
+      case mode
+        when MODE_UNLIMITED
+          _('Unlimited')
+        when MODE_WHITELIST
+          _('Whitelist')
+        when MODE_BLACKLIST
+          _('Blacklist')
+        else
+          _('Unknown listing mode')
+      end
+    end
+
+    class ListCommand < HammerCLIForeman::ListCommand
+      output do
+        field :id, _('Id')
+        field :name, _('Name')
+        field :_interval, _('Interval')
+        field :_status, _('Status')
+        field :last_report_at, _('Last Report At'), Fields::Date
+      end
+
+      def extend_data(conf)
+        conf['_interval'] = VirtWhoConfig.format_interval(conf['interval'])
+        conf['_status'] = VirtWhoConfig.format_status(conf['status'])
+        conf
+      end
+
+      build_options
+    end
+
+    class InfoCommand < HammerCLIForeman::InfoCommand
+      output do
+        label _('General information') do
+          field :id, _('Id')
+          field :name, _('Name')
+          field :hypervisor_type, _('Hypervisor type')
+          field :hypervisor_server, _('Hypervisor server')
+          field :hypervisor_username, _('Hypervisor username')
+          field :hypervisor_password, _('Hypervisor password')
+          field :_status, _('Status')
+        end
+        label _('Schedule') do
+          field :_interval, _('Interval')
+          field :last_report_at, _('Last Report At'), Fields::Date
+        end
+        label _('Connection') do
+          field :satellite_url, _('Satellite server')
+          field :hypervisor_id, _('Hypervisor ID')
+          field :_filtering_mode, _('Filtering')
+          field :blacklist, _('Filtered hosts'), Fields::Field, :hide_blank => true
```

Comment:
I think the translations need to be swapped with whitelist which filters hosts to be parsed, while blacklist excludes hosts